core/src/main/java/org/mule/api/transport/ErrorHandlingReplyToHandlerDelegate.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.api.transport;
+
+import org.mule.api.MessagingException;
+import org.mule.api.MuleEvent;
+import org.mule.api.MuleException;
+import org.mule.api.MuleMessage;
+import org.mule.api.exception.MessagingExceptionHandler;
+import org.mule.api.CompletionHandler;
+
+/**
+ * {@link org.mule.api.transport.ReplyToHandler} implementation that uses a {@link org.mule.api.exception
+ * .MessagingExceptionHandler}to handle errors before delegating to the delgate ReplyToHandler instance.
+ * <p/>
+ * Invocations of {@link #processReplyTo(org.mule.api.MuleEvent, org.mule.api.MuleMessage, Object)} are passed straight
+ * through to the delegate ReplyToHandler where as invocations of
+ * {@link #processExceptionReplyTo(org.mule.api.MuleEvent, org.mule.api.MessagingException, Object)} may result in a
+ * delegation to either {@link #processReplyTo(org.mule.api.MuleEvent, org.mule.api.MuleMessage, Object)} or
+ * {@link #processExceptionReplyTo(org.mule.api.MuleEvent, org.mule.api.MessagingException, Object)} depending on the
+ * result of {@link org.mule.api.MessagingException#handled()} after the MessagingExceptionHandler has been invoked.
+ */
+public class ErrorHandlingReplyToHandlerDelegate implements ReplyToHandler
```

Comment:
is this a ErrorHandlingReplyToHandlerDelegate or a ErrorHandlingReplyToHandlerWrapper?